linux/s2e.so/s2e.c

```diff
@@ -73,39 +73,76 @@ typedef struct _procmap_entry_t {
     const char *name;
 } procmap_entry_t;
 
-// Returns each executable line of /proc/self/maps
-// as an array of entries. The last one is null.
+static inline int read_procmap_entry(char *line, uintptr_t *base, uintptr_t *limit, unsigned *inode, char *path) {
+    return sscanf(line, "%" SCNxPTR "-%" SCNxPTR " %*c%*c%*c%*c %*x %*s %d %127[^\n]", base, limit, inode, path);
+}
+
+// Returns each executable line of /proc/self/maps as an array of entries. The last one is NULL.
 static procmap_entry_t *load_process_map(void) {
     FILE *maps = fopen("/proc/self/maps", "r");
     if (!maps) {
         return NULL;
     }
 
+    int matches;
     char line[256], path[128];
     procmap_entry_t *result = NULL;
     procmap_entry_t current_entry;
     unsigned nb_entries = 0;
-    char executable;
-    while (fgets(line, sizeof(line), maps)) {
-        int matches = sscanf(line, "%" SCNxPTR "-%" SCNxPTR " %*c%*c%c%*c %*x %*s %*d %127[^\n]", &current_entry.base,
-                             &current_entry.limit, &executable, path);
+
+    uintptr_t start_addr, end_addr;
+    unsigned current_inode = 0, previous_inode = 0;
+
+    // Read /proc/self/maps until we can initialize previous_inode to a non-zero inoda and then initialize the first
```

Comment:
inoda => inode?

Reply:
It would be good to add the format of /proc/self/maps as a comment in the code.
Basically, paste your commit msg into the code.